### PR TITLE
Slide transition

### PR DESCRIPTION
**Pull request by 0xDeon**
I created a smooth slide transition on each of the pages menu tab

# Pull Request type
Feature

## Summary by CodeRabbit

- **New Features**
	- Introduced new components: `OdTab` and `SlidingTab` for improved tab navigation.
	- Integrated custom tab components in various parts of the application.
- **Bug Fixes**
	- Adjusted alignment and background color properties in styles for better UI consistency.
- **Chores**
	- Updated various dependencies to newer versions, including `@mui/x-date-pickers`, `dayjs`, `jwt-decode`, and `react-datepicker`.
- **Refactor**
	- Simplified tab handling logic and improved import clarity across multiple files.

### CODE REVIEW

File: package-lock.json
Comment (ayushtom) [inline package-lock.json:1]: can you please remove this change if there no changes in the package.json

**Review by ayushtom — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (0xDeon) [reply package-lock.json:1]: All done, please review @ayushtom

**Comment by ayushtom** [comment]
https://github.com/starknet-id/starknet.quest/assets/41674634/757fe6b5-9b9b-4348-9379-7063121bacf6

This is not looking anywhere close to what i shared :)
Can you please work on this more and implement a working version of this?

**Comment by 0xDeon** [comment]
@ayushtom I'll get right on it, almost done with the changes

**Comment by 0xDeon** [comment]
@ayushtom All done, please review

**Comment by ayushtom** [comment]
@DanielEmmanuel1 your build is failing. I deployed it yesterday as you can see. Please resolve it

**Comment by 0xDeon** [comment]
> @DanielEmmanuel1 your build is failing. I deployed it yesterday as you can see. Please resolve it
@ayushtom  Issue resolved, pls review.

**Comment by ayushtom** [comment]
Screenshot 2024-06-03 at 11 32 29 AM

Good job on the animation!

i think dynamic width is not implemented yet which is why this looks too big on the sides here. Also can you make the tabs of the of the same styling like they were before?
Also you can use a shade of gray or something which you feel looks nice for the animation here for hovered state. The white one doesn't look that nice.

**Comment by ayushtom** [comment]
https://github.com/starknet-id/starknet.quest/pull/701#issuecomment-2144346751

Apart from hover color
the review still applies. Please resolve this

**Comment by 0xDeon** [comment]
@ayushtom for clarity, the previous problems were the width size and and active state style, on telegram u said on over should be gray and on click should be white which i beleieve is what it is right now, except I'm mixing something up

**Comment by ayushtom** [comment]
Screenshot 2024-06-14 at 3 57 50 AM
Breaks for me on the deployment 

Also please resolve the conflicts

File: app/[addressOrDomain]/page.tsx [hunk 1]
```diff
@@ -310,47 +315,31 @@ export default function Page({ params }: AddressOrDomainProps) {
...
-              }}
-              label={`To claim (${
-                claimableQuests ? claimableQuests.length : 0
-              })`}
-              {...a11yProps(1)}
-            />
-          </Tabs>
+          <nav className="pb-4" id="navTabWrapper" style={{borderBottom: "0.5px solid rgba(224, 224, 224, 0.3)"}}>
+            <div className="dash-box">
+              <OdTab
```
Comment (ayushtom) [inline app/[addressOrDomain]/page.tsx]: can you rename this to `SlidingTab`

File: app/[addressOrDomain]/page.tsx [hunk 2]
```diff
@@ -310,47 +315,31 @@ export default function Page({ params }: AddressOrDomainProps) {
...
+                  claimableQuests ? claimableQuests.length : 0
+                })`}
+                names={`w-fit py-2 px-2 nav-item2 text-center ${
+                  tabIndex === 1 ? "active" : "text-white"
+                }`}
+                setActive={(event) => {handleChangeTab(1); moveSpanActive(event, ".dash-box")}}
+                prep={1}
+                mouseLeave={(event) => handleMouseLeave(event, ".dash-box")}
+              />
+              <span/>
```
Comment (ayushtom) [inline app/[addressOrDomain]/page.tsx:339]: why empty span here ?

File: app/quest/[questPage]/page.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ type Props = {
 
 export async function generateMetadata(
   { params }: Props,
-  parent: ResolvingMetadata
+  // parent: ResolvingMetadata
```
[the remark this answers is not in the document]
Comment (ayushtom) [reply app/quest/[questPage]/page.tsx]: why this change?

File: components/UI/navbar.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ import Button from "./button";
 import { useConnect, useAccount, useDisconnect } from "@starknet-react/core";
 import ModalMessage from "./modalMessage";
 import { useDisplayName } from "@hooks/displayName.tsx";
-import { useDomainFromAddress } from "../../hooks/naming";
+// import { useDomainFromAddress } from "../../hooks/naming";
```
Comment (ayushtom) [inline components/UI/navbar.tsx]: why removing this?

File: package-lock.json
Comment (ayushtom) [inline package-lock.json:1]: do we add any new packages?

File: package.json [hunk 5]
```diff
@@ -20,7 +20,7 @@
     "@react-three/drei": "^9.80.3",
     "@react-three/fiber": "^8.13.6",
     "@starknet-react/chains": "^0.1.6",
-    "@starknet-react/core": "^2.8.0",
+    "@starknet-react/core": "^2.8.2",
```
Comment (ayushtom) [inline package.json]: please revert this change if you aren't using it in your code

**Review by ayushtom — CHANGES_REQUESTED** [review]

**Comment by 0xDeon** [comment]
All done, all changes made, also the span is the white background that make the transistion from each item tab @ayushtom

**Review by fricoben — CHANGES_REQUESTED** [review]
Seems to add a lot of unnecessary code and complexity. 

I'll close the PR and but still reward on OD.

**Pull request closed without merging**